src/main/java/Controller/RegisterController.java

```diff
@@ -0,0 +1,41 @@
+package Controller;
+
+import InputBoundary.loginInputBoundary;
+import InputBoundary.RegisterArtistInputBoundary;
+import InputBoundary.RegisterInputBoundary;
+import InputData.RegisterInputData;
+import InputData.RegisterArtistInputData;
+import InputData.loginInputData;
+
+
+/**
+ * Controller for login and register use cases.
+ */
+public class RegisterController {
+    private final RegisterInputBoundary registerInteractor;
+    private final RegisterArtistInputBoundary artistInteractor;
+    private final RegisterInputData registerInputData;
```

Comment:
Same point as in loginController -- registerInputData should be temporary.

Plus, since the Controller is final in the ViewModel, this would also never change -- meaning that, if a user were to incorrectly register or log in, their incorrect attempt would be the final registerInputData and incapable of being replaced.